identity-service/src/utils/withdrawUSDCInstructionsHelpers.js

```diff
@@ -0,0 +1,68 @@
+const {
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  TOKEN_PROGRAM_ID
+} = require('@solana/spl-token')
+const { Keypair } = require('@solana/web3.js')
+const config = require('../config')
+
+const solanaClaimableTokenProgramAddress = config.get(
+  'solanaClaimableTokenProgramAddress'
+)
+
+const allowedProgramIds = [
+  solanaClaimableTokenProgramAddress,
+  /* secp */ 'KeccakSecp256k11111111111111111111111111111'
+]
+
+const isRelayAllowedProgram = (instructions) => {
+  for (const instruction of instructions) {
+    if (!allowedProgramIds.includes(instruction.programId)) {
+      return false
+    }
+  }
+  return true
+}
+
+const checkCreateAccountInstruction = (instruction) => {
+  const isCreateInstruction =
```

Comment:
I think we need to check the instruction index to ensure it's an `open`, like you did in the close